awx/ui_next/src/screens/Inventory/InventoryGroups/InventoryGroups.jsx

```diff
@@ -1,10 +1,259 @@
+    isModalOpen,
+    toggleModal,
+  };
+};
+
+function InventoryGroups({ i18n, location, match }) {
+  const [actions, setActions] = useState(null);
+  const [contentError, setContentError] = useState(null);
+  const [deletionError, setDeletionError] = useState(null);
+  const [groupCount, setGroupCount] = useState(0);
+  const [groups, setGroups] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [selected, setSelected] = useState([]);
+  const { isModalOpen, toggleModal } = useModal();
+
+  const inventoryId = match.params.id;
+
+  const fetchGroups = (id, queryString) => {
+    const params = parseQueryString(QS_CONFIG, queryString);
+    return InventoriesAPI.readGroups(id, params);
+  };
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const [
+          {
+            data: { count, results },
+          },
+          {
+            data: { actions: optionActions },
+          },
+        ] = await Promise.all([
+          fetchGroups(inventoryId, location.search),
+          InventoriesAPI.readGroupsOptions(inventoryId),
+        ]);
+
+        setGroups(results);
+        setGroupCount(count);
+        setActions(optionActions);
+      } catch (error) {
+        setContentError(error);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    fetchData();
+  }, [inventoryId, location]);
+
+  const handleSelectAll = isSelected => {
+    setSelected(isSelected ? [...groups] : []);
+  };
+
+  const handleSelect = row => {
+    if (selected.some(s => s.id === row.id)) {
+      setSelected(selected.filter(s => s.id !== row.id));
+    } else {
+      setSelected(selected.concat(row));
+    }
+  };
+
+  const renderTooltip = () => {
+    const itemsUnableToDelete = selected
+      .filter(cannotDelete)
+      .map(item => item.name)
+      .join(', ');
+
+    if (selected.some(cannotDelete)) {
+      return (
+        <div>
+          {i18n._(
+            t`You do not have permission to delete the following Groups: ${itemsUnableToDelete}`
+          )}
+        </div>
+      );
+    }
+    if (selected.length) {
+      return i18n._(t`Delete`);
+    }
+    return i18n._(t`Select a row to delete`);
+  };
+
+  const promoteGroups = list => {
+    const promotePromises = Object.keys(list)
+      .filter(groupId => list[groupId] === 'promote')
+      .map(groupId => InventoriesAPI.promoteGroup(inventoryId, +groupId));
+
+    return Promise.all(promotePromises);
+  };
+
+  const deleteGroups = list => {
+    const deletePromises = Object.keys(list)
+      .filter(groupId => list[groupId] === 'delete')
+      .map(groupId => GroupsAPI.destroy(+groupId));
+
+    return Promise.all(deletePromises);
+  };
+
+  const handleDelete = async list => {
+    setIsLoading(true);
+
+    try {
+      await Promise.all([promoteGroups(list), deleteGroups(list)]);
+    } catch (error) {
+      setDeletionError(error);
+    } finally {
+      toggleModal();
+      setSelected([]);
+
+      try {
```

Comment:
Right, I couldn't think of a better way to handle this. I could move the nested try/catch into a separate function since the nested catch doesn't need to bubble up. What do you think?